tools/compiler/main.rs

```diff
@@ -32,60 +32,72 @@ enum Embedding {
 #[derive(Parser)]
 #[command(author, version, about, long_about = None)]
 struct Cli {
-    /// Set output format
-    #[arg(short = 'f', long = "format", default_value = "cpp", action)]
+    /// Set the output format for generated code.
+    /// Possible values: 'cpp' for C++ code or 'rust' for Rust code.
+    #[arg(short = 'f', long = "format", default_value = "cpp")]
     format: generator::OutputFormat,
 
-    /// Include path for other .slint files
-    #[arg(short = 'I', name = "include path", number_of_values = 1, action)]
+    /// Specify include paths for imported .slint files or image resources.
+    /// This is used for including external .slint files or image resources referenced by '@image-url'.
+    #[arg(short = 'I', name = "include path", number_of_values = 1)]
     include_paths: Vec<std::path::PathBuf>,
 
-    /// The argument should be in the format `<library>=<path>` specifying the
-    /// name of the library and the path to the library directory or a .slint
-    /// entry-point file.
-    #[arg(short = 'L', name = "library path", number_of_values = 1, action)]
+    /// Define library paths in the format `<library>=<path>`.
+    /// This can point to either a library directory or a .slint entry-point file.
+    #[arg(short = 'L', name = "library path", number_of_values = 1)]
     library_paths: Vec<String>,
 
-    /// Path to .slint file ('-' for stdin)
-    #[arg(name = "file", action)]
+    /// Specify the path to the main .slint file to compile.
+    /// Use '-' to read from stdin.
+    #[arg(name = "file")]
     path: std::path::PathBuf,
 
-    /// The style name ('native' or 'fluent')
-    #[arg(long, name = "style name", action)]
+    /// Set the style for the UI (e.g., 'native' or 'fluent').
+    #[arg(long = "style-name")]
     style: Option<String>,
 
-    /// The constant scale factor to apply for embedded assets and set by default on the window.
-    #[arg(long, name = "scale factor", action)]
+    /// Apply a constant scale factor to embedded assets, typically for high-DPI displays.
+    /// This scale factor is also applied to the window by default.
+    #[arg(long = "scale-factor")]
     scale_factor: Option<f64>,
 
-    /// Generate a dependency file
-    #[arg(name = "dependency file", long = "depfile", number_of_values = 1, action)]
+    /// Generate a dependency file for build systems like CMake or Ninja.
+    /// This file is similar to the output of `gcc -M`.
+    #[arg(long = "depfile", name = "dependency file", number_of_values = 1)]
     depfile: Option<std::path::PathBuf>,
 
-    /// Declare which resources to embed
-    #[arg(long, name = "value", value_enum)]
+    /// Declare which resources to embed into the final output.
+    /// Use the enum `Embedding` to specify resource types.
```

Comment:
This reference seems not so useful in the command line interface. I'd either omit it or (better) list the options. Unless clap shows them?

Reply:
right, this shouldn't be there. Clap auto-detect options from enum.